docker-java/src/test/java/com/github/dockerjava/cmd/AttachContainerCmdIT.java

```diff
@@ -281,6 +282,57 @@ public void onComplete() {
         }
     }
 
+    /**
+     * The process in Docker container should receive EOF when reading its
+     * stdin after the corresponding stream is closed on docker-java side in
+     * case when the container has been created with StdinOnce flag set to
+     * true.
+     */
+    @Test
+    public void closeStdinWithStdinOnce() throws Exception {
+        Assume.assumeTrue("supports stdin attach", getFactoryType().supportsStdinAttach());
+        Assume.assumeFalse("not fixed for OkHttp", getFactoryType().equals(FactoryType.OKHTTP));
+
+        DockerClient dockerClient = dockerRule.getClient();
+
+        CreateContainerResponse container = dockerClient.createContainerCmd(DEFAULT_IMAGE)
+            .withStdinOpen(true)
+            .withStdInOnce(true)
+            .withCmd("sh", "-c", "read -r line; echo Done")
+            .exec();
+
+        CountDownLatch done = new CountDownLatch(1);
+        PipedOutputStream stdin = new PipedOutputStream();
+
+        try (
+            ResultCallback.Adapter<Frame> resultCallback = dockerClient.attachContainerCmd(container.getId())
```

Comment:
Hi @bsideup 
It's not related to the current PR, but probably you can help me with my question

Is it a best practice to create ResultCallbackTemplate in a try-catch-resource block? 
After checking the default implementation of `ResultCallbackTemplate`, it should automatically close the docker client into the method `onComplete`. 

```java
// This callback will write container's logs to the application logs
DockerClient dockerClient1 = openDefaultClient();
      dockerClient1
          .logContainerCmd(containerId)
          .withStdOut(true)
          .withStdErr(true)
          .withTimestamps(true)
          .withFollowStream(true)
          .exec(containerLogsCallback(request, containerId));

      // This one will check container events and we close it manually when get the `event.getStatus()` equals to `destroy`
      DockerClient dockerClient2 = openDefaultClient();
      dockerClient2.eventsCmd()
          .withContainerFilter(containerId)
          .exec(new ContainerEventsCallback(containerId));
```